### PR TITLE
ENH accelerate load_concat_dataset by pickling unloaded raw object

### PR DESCRIPTION
**Pull request by tomMoral**

### CODE REVIEW

File: braindecode/datautil/serialization.py [hunk 1]
```diff
@@ -114,12 +116,33 @@ def _load_signals_and_description(path, preload, is_raw, ids_to_load=None):
 
 
 def _load_signals(fif_file, preload, is_raw):
```
Comment (sliwy) [inline braindecode/datautil/serialization.py:118]: maybe we could add type hints when modifying this function? reading the code could be much easier then

File: braindecode/datautil/serialization.py [hunk 2]
```diff
@@ -114,12 +116,33 @@ def _load_signals_and_description(path, preload, is_raw, ids_to_load=None):
...
+
+    # Reading the raw file from pickle if it has been save before.
+    # The pickle file only contain the raw object without the data.
+    pkl_file = fif_file.with_suffix(".pkl")
+    if pkl_file.exists():
```
Comment (sliwy) [inline braindecode/datautil/serialization.py:123]: maybe we can add some logging that we'are loading from pickle? I imagine situation in which you modify raw files and you're stuck with old pickle files.

File: braindecode/datautil/serialization.py [hunk 3]
```diff
@@ -146,6 +169,8 @@ def load_concat_dataset(path, preload, ids_to_load=None, target_name=None,
     -------
     concat_dataset: BaseConcatDataset of BaseDatasets or WindowsDatasets
     """
+    path = Path(path)
```
Comment (sliwy) [inline braindecode/datautil/serialization.py:172]: I would allow in the parameters to pass `pathlib.Path` also as path parameter.

Comment (tomMoral) [reply braindecode/datautil/serialization.py:181]: Did this! :)

File: braindecode/datautil/serialization.py (hunk 1 above)
Comment (tomMoral) [reply braindecode/datautil/serialization.py:118]: Maybe for another PR.

**Comment by tomMoral** [comment]
Thanks for the review @sliwy ! Appreciated.

**Pull request merged**